functionary/core/utils/parameter.py

```diff
@@ -96,6 +100,8 @@ def validate_parameters(parameters: dict, instance: Union["Function", "Workflow"
     """
     try:
         pydantic_model = _get_pydantic_model(instance)
-        _ = pydantic_model(**_serialize_json_parameters(parameters, instance))
+        validate(
```

Comment:
Silly as it seems, your commented out test could pass if you use both the pydantic and jsonschema validation here, which you can do by calling `.schema()` on an instance of the model rather than the model class and then catching both types of `ValidationError`:

```python
    try:
        pydantic_model = _get_pydantic_model(instance)
        serialized_parameters = _serialize_json_parameters(parameters, instance)

        jsonschema.validate(
            serialized_parameters, pydantic_model(**serialized_parameters).schema()
        )
    except (ValidationError, jsonschema.ValidationError, json.JSONDecodeError) as exc:
        raise DjangoValidationError(exc)
```